airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java

```diff
@@ -316,6 +318,48 @@ public JdbcDatabase createDatabase(final JsonNode config) throws SQLException {
     return database;
   }
 
+  /**
+   * Retrieves connection_properties from config and also validates if custom
+   * jdbc_url parameters overlap with the default properties
+   *
+   * @param config A configuration used to check Jdbc connection
+   * @return A mapping of connection properties
+   */
+  protected Map<String, String> getConnectionProperties(final JsonNode config) {
+    final Map<String, String> customProperties = JdbcUtils.parseJdbcParameters(config, JdbcUtils.JDBC_URL_PARAMS_KEY);
+    final Map<String, String> defaultProperties = getDefaultConnectionProperties(config);
+    assertCustomParametersDontOverwriteDefaultParameters(customProperties, defaultProperties);
+    return MoreMaps.merge(customProperties, defaultProperties);
+  }
+
+  /**
+   * Validates for duplication parameters
+   *
+   * @param customParameters custom connection properties map as specified by each Jdbc source
+   * @param defaultParameters connection properties map as specified by each Jdbc source
+   * @throws IllegalArgumentException
+   */
+  private void assertCustomParametersDontOverwriteDefaultParameters(final Map<String, String> customParameters,
+      final Map<String, String> defaultParameters) {
+    for (final String key : defaultParameters.keySet()) {
+      if (customParameters.containsKey(key) && !Objects.equals(customParameters.get(key), defaultParameters.get(key))) {
+        throw new IllegalArgumentException("Cannot overwrite default JDBC parameter " + key);
```

Comment:
Question: when Airbyte tries to run a sync job, will this exception be thrown if a user has created a connection with an older version of the connector that did not perform this validation and then upgraded to the version of connector that performs this validation?

Reply:
Yes, there will be an exception thrown upon upgrading to this new version where validation would exist. Based on that, it seems like I should note this commit will potentially cause breaking changes. 

I'm curious to know if when the validation check was added to the `AbstractJdbcDestination` did we notify customers ahead of time as well because the validation check did not originally exist to the best of my knowledge. @girarda it seems you were the person that last touched this `assertCustomParametersDontOverwriteDefaultParameters` method in `AbstractJdbcDestination`, can you confirm if we notified customers that their `jdbc_url` parameters could break or how did we approach this?

Reply:
We didn't notify the customers because it wasn't yet possible to set `jdbc_url` parameters when we [added that check](https://github.com/airbytehq/airbyte/pull/10421/files#diff-f223085c9a35f32d96680afc95c5bb7a133d031168ba6e7256fbecc9a709809b)

Reply:
Given that we would like to have feature parity of validation between `Sources` and `Destinations` it seems that this is a concern that I would defer the decision to someone who can make that executive call. 

My thoughts would be, while it may be unfortunate that users will have breaking changes, the scenario for a breaking change is mostly if `Postgres` users have conflicting `ssl` settings in their `jdbc_url` in which case they may even already have unexpected results since the `jdbc_url` parameters may be overwriting our own custom `ssl` configurations. However, I cannot say with 100% certainty that there might not be other scenarios that could break a customer's `jdbc_url` parameters so will want to proceed with caution

Reply:
agreed with Subodh